libs/ui/src/hooks/useHardware.ts

```diff
@@ -0,0 +1,175 @@
+import { useEffect, useState } from 'react';
+import {
+  isCardReader,
+  Hardware,
+  isAccessibleController,
+  isPlustekVTM300Scanner,
+  isFujitsuScanner,
+} from '@votingworks/utils';
+import { map } from 'rxjs/operators';
+import { LogEventId, Logger } from '@votingworks/logging';
+import { usePrevious } from '..';
+
+export interface UseHardwareProps {
+  hardware: Hardware;
+  logger: Logger;
+}
+
+export interface UseHardwareResult {
+  hasCardReaderAttached: boolean;
+  hasAccessibleControllerAttached: boolean;
+  hasPlustekScannerAttached: boolean;
+  hasFujitsuScannerAttached: boolean;
```

Comment:
Do we want the brand names here? Or should we do something like this?

```suggestion
  hasPrecinctScannerAttached: boolean;
  hasBatchScannerAttached: boolean;
```